docs: Add GCP service account guide (#547)
devopness/devopness: docs/cloud-providers/add-cloud-providers-credentials.md

    featured:
---

# Generating Cloud Providers Credentials
## GCP Service Account Credentials

> 1. After creating your project, got to GCP IAM manager and select the `Service Accounts` option on the left menu

Reviewer: > After creating your project
Devopness project? GCP project?

Shouldn't we be more direct to the user on the basic requirements?
If a user does not even have a project on her GCP account, very likely the person is not yet ready to provision resources on their account not even manually.

Maybe something like:
```
Access your GCP project or [create a new project on GCP](https://cloud.google.com/resource-manager/docs/creating-managing-projects)
```
?
This way we give users sources for getting into deeper details and remove from us the responsibility of training users on GCP specifics.

Author: Suggestion accepted: https://github.com/devopness/devopness/blob/614e3070a24c883a787e519c4821679fe2bbfc67/docs/cloud-provider-credentials/get-gcp-service-account-credentials.md?plain=1#L17

> 2. If you already have a service account, jump to step 5

Reviewer: Perhaps only deviate from the regular step by step flow when something is missing?
```suggestion
2. If you do not have a service account, follow [instructions on GCP docs to create a service account key](https://cloud.google.com/iam/docs/creating-managing-service-account-keys)
```

> 2. Go to GCP IAM manager and select the `Service Accounts` option on the left menu

Reviewer 2: ```suggestion
2. Select the `Service Accounts` option under `IAM & Admin` on the navigation menu
```

Author: How about this:
https://github.com/devopness/devopness/blob/614e3070a24c883a787e519c4821679fe2bbfc67/docs/cloud-provider-credentials/get-gcp-service-account-credentials.md?plain=1#L29

3. If you do not have a service account, follow [instructions on GCP docs to create a service account key](https://cloud.google.com/iam/docs/creating-managing-service-account-keys)

> 4. Provide the `Cumpute Instance Admin (v1)` and `Service Account Manager` roles to your service account

Reviewer 2: ```suggestion
4. Provide the `Compute Instance Admin (v1)` and `Service Account Manager` roles to your service account
```

Author: Fixed
https://github.com/devopness/devopness/blob/614e3070a24c883a787e519c4821679fe2bbfc67/docs/cloud-provider-credentials/get-gcp-service-account-credentials.md?plain=1#L42
